src/os_win/os_futex.c

```diff
@@ -0,0 +1,51 @@
+/*-
+ * Copyright (c) 2024-present MongoDB, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+/*
+ * __wt_futex_wait --
+ *     Wait on the futex. The timeout is in microseconds and MUST be greater than zero.
+ */
+int
+__wt_futex_wait(
+  volatile WT_FUTEX_WORD *addr, WT_FUTEX_WORD expected, time_t usec, WT_FUTEX_WORD *wake_valp)
+{
+    DWORD msec;
+    DWORD windows_error;
+    bool retval;
+
+    WT_ASSERT(NULL, usec > 0);
+
+    msec = (DWORD)WT_MAX(1, usec / 1000);
+    retval = WaitOnAddress(addr, &expected, sizeof(WT_FUTEX_WORD), msec);
+    if (retval == TRUE) {
+        *wake_valp = *addr;
```

Comment:
Should this be an atomic load like for the other platforms? If not, maybe a comment would be helpful. (Is it not necessary because we currently support Windows only on x86? Windows can now also work on ARM, so we should at least leave a comment in case we would support it in the future.)

Reply:
@marcbutler I'm also curious about this - my guess is different to Peter's though ;-) Is it due to MSVC's poor C11 atomics support? If so, we can probably do another macro like the `futex_atomic_store` in the tests. I have to admit I have no idea what the equivalent `InterlockedXYZ` intrinsic would be.

Reply:
There is no `InterlockedRead` in windows API. I've seen it synthesized with `InterlockedOr(ptr, 0)` but I think that is both convoluted and confusing. The x86 TSO (total store ordering) model ensures that every read will receive the value of the last store to that location and I assume this API reflects that heritage. @pmacko86 is correct in that if we move to ARM this is an issue. That said msvc supports the atomic API for C++ but not C, which is the rub here.
Shall I can create a pre-emptive ticket to bring attention to this?

Reply:
The absence of `InterlockedRead` is unfortunate 😞

I don't have a strong feelings either way, but here are some thoughts if it helps... If we add support for ARM, I'd like to believe that we would also carefully review Windows-specific code, so hopefully the comment would be overlooked. I'm a bit hesitant to create a ticket that we wouldn't be able to address for a few years, and we could run into the same problem that the ticket may get overlooked—but hopefully the probability of that would be less due to a `FIXME` comment in the code.

Alternatively, it just occurred to me that maybe we could just use the appropriate barrier from `gcc.h` in this case? It would be a bit ugly, but hopefully at least better than `InterlockedOr`?

I could really go either way on this—I'm happy to leave this up to your preference.

Reply:
I'm comfortably leaving it as-is, because it matches Microsoft's own [code example](https://learn.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-waitonaddress):
```
ULONG g_TargetValue; // global, accessible to all threads
ULONG CapturedValue;
ULONG UndesiredValue;

UndesiredValue = 0;
CapturedValue = g_TargetValue;
while (CapturedValue == UndesiredValue) {
      WaitOnAddress(&g_TargetValue, &UndesiredValue, sizeof(ULONG), INFINITE);
      CapturedValue = g_TargetValue;
}
```
I don't think there is a sure fire way to ensure the comment is seen at the right time, but I would hope that taking on a new platform would trigger a preparatory audit. And in that case this feels like a reasonable flag.

Reply:
Sounds good—let's leave it as-is then.

Thank you for checking!